### PR TITLE
Params evalkey removal and changes to unit tests

### PR DESCRIPTION
**Pull request by dsuponitskiy**
1. Removed unused parameter EvalKey from LevelReduceInternal() and LevelReduceInternalinPlace()
2. Refactored UnitTestEvalMult.cpp, UnitTestCKKSrnsAutomorphism.cpp, UnitTestBGVrnsAutomorphism.cpp and UnitTestBGVrnsAdvancedSHE.cpp to have tests "data driven" in these files
3. Added additional test cases for BGVRNS and CKKSRNS

### CODE REVIEW

**Review by jbates-duality — APPROVED** [review]
The intent of this patch is accidentally twofold.

1) remove EvalKey param from various method calls.
2) fairly major re-org/refactor of unit tests.

Code is clean, well-formed, and easy to look at.  I checked out the branch, built portable, and ran all unittests with success.

I am approving now.

**Review by yspolyakov — APPROVED** [review]
Can be merged after PR #57 is merged

**Pull request merged**
